### DISCUSSION TITLE
Single `HttpClient` with several circuit breaker

### DISCUSSION THREAD

**Discussion by ghost** [App-vNext/Polly#1802] [Q&A] [answered]
I'm using a single `HttpClient` to perform http calls to several backends with a circuit breaker policy: 

```cs
builder.Services
    .AddHttpClient("Default")
    .AddResilienceHandler("default", b =>
    {
        b.AddCircuitBreaker(new()
        {
            BreakDuration = TimeSpan.FromSeconds(5),
            FailureRatio = 0.1,
            MinimumThroughput = 100,
            SamplingDuration = TimeSpan.FromSeconds(30),
        });
    });
```

Is there any way to have a different circuit breaker for each backend that is consumed (depending on the hostname for example)? Or would it be better to create different `HttpClient` and apply the circuit breaker to each one?

**Comment by martincostello** [comment]
Without a more complete example of exactly what you're doing, I would have thought that having a different HttpClient for each endpoint with their own circuit breaker would be the easiest way to go about things.

**Comment by martintmk** [comment, marked as answer]
@joegoldman2 

> Is there any way to have a different circuit breaker for each backend that is consumed

This will work with a slight modification to your sample:

``` csharp
builder.Services
    .AddHttpClient("Default")
    .AddResilienceHandler("default", b =>
    {
        b.AddCircuitBreaker(new()
        {
            BreakDuration = TimeSpan.FromSeconds(5),
            FailureRatio = 0.1,
            MinimumThroughput = 100,
            SamplingDuration = TimeSpan.FromSeconds(30),
        });
    })
    .SelectPipelineByAuthority();
```

Notice the use of `SelectPipelineByAuthority`. This tells the handler to cache the pipeline by authority (scheme + host + port) extracted from request message. Effectively, this assigns a unique instance of pipeline with circuit breaker to each authority and does exactly what you want. 
 
``` csharp
httpClient.GetAsync("https://A"); // creates and caches the pipeline with "https://A" key

httpClient.GetAsync("https://B"); // creates and caches the pipeline with "https://B" key
```
